### DISCUSSION TITLE
PropertyChanged events are null on macOS

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9514] [Q&A]
[_From Emma111#5878 on Discord:_](https://discord.com/channels/372137812037730304/550416151172087808/1009006170054344704)

Hi, I have a UWP application I'm trying to port to MacOS with Uno, it works fine in UWP but on Mac PropertyChanged events are null when they're raised, so I can't update views. It has a few other niggles, but this is what's really stumped me. Maybe I need to set data context? Or add partial to more classes? I added partial to the notifier my viewmodels inherit from. Any help is greatly appreciated!

**Comment by jeromelaban** [comment]
Would you have an example of code or simple project you can share? There may be something that is missing. A few other things, have you tried with a `dev` prerelease version of Uno, and have you tried on another platform (like Skia.Gtk, which also runs on mac) ?

**Reply by Tapir28** [reply]
Hi, so I've been doing a bit more digging, I now think propertychanged being null is just because it's never subscribed. xBind works on mac 
When I test just simple binds to the view model.

But the program dynamically creates views with a view factory for it's views. Views returned by the view factory CreateView function are bound to the content of xaml elements which contain them, such as the splitview I'm debugging like so:

```
<SplitView
			x:Name="splitView"
			Grid.Row="1"
			Background="{StaticResource appBackground}"
			DisplayMode="CompactOverlay"
			PaneOpening="SplitView_PaneOpening"
			PaneClosing="SplitView_PaneClosing"
			OpenPaneLength="270"
			Content="{x:Bind m_viewFactory.CreateView(m_viewModel.MainViewType), Mode=OneWay}">
```

On windows this is fine, CreateView runs and returns a view no problem.

But on Mac it just doesn't execute. The program builds and runs, but wherever there should be generated views there are empty spaces with the content unpopulated.

I haven't tried for skia, or the dev release, I'll give those approaches a whirl.
